lib/Frontend/FrontendOptions.cpp

```diff
@@ -283,6 +288,7 @@ bool FrontendOptions::canActionEmitObjCHeader(ActionType action) {
   case ActionType::EmitAssembly:
   case ActionType::EmitObject:
   case ActionType::EmitImportedModules:
+  case ActionType::DumpTypeInfo:
```

Comment:
This definitely seems odd. Just because we got past typechecking doesn't mean it makes sense to have supplementary outputs. I'd change most of these `canActionEmit*` methods to return false, like the other dump options.